lib/common.js

```diff
@@ -23,13 +23,17 @@ exports.runDoctests = function(doctest, program) {
     ]));
     process.exit (1);
   }
-  Promise.all (program.args.map (function(path) {
-    return (doctest (path, program)).then (function(results) {
-      return results.reduce (function(status, tuple) {
-        return tuple[0] ? status : 1;
-      }, 0);
+  program.args.reduce (function(promise, path) {
+    return promise.then (function(statuses) {
+      return doctest (path, program).then (function(results) {
```

Comment:
I slightly prefer the following:

```suggestion
      return (doctest (path, program)).then (function(results) {
```